src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -403,6 +441,99 @@ inline std::array< int, 3 > getijk( const DeckRecord& record ) {
     }};
 }
 
+
+inline void keywordCL( SummaryConfig::keyword_list& list,
+                       const ParseContext& parseContext,
+                       ErrorGuard& errors,
+                       const DeckKeyword& keyword,
+                       const Schedule& schedule ,
+                       const GridDims& dims)
+{
+
+    auto make_node = [&list, &keyword](const std::string& wname, const Connection& conn) {
+        auto node = SummaryConfigNode{keyword.name(), SummaryConfigNode::Category::Connection, keyword.location()};
+        node.parameterType( parseKeywordType(keyword.name()) );
+        node.isUserDefined( is_udq(keyword.name()) );
+        node.number( 1 + conn.global_index());
+        node.namedEntity( wname );
+
+        list.push_back( std::move(node) );
+    };
```

Comment:
This is more expensive than it needs to be because you end up parsing out the same information for each node.  If you use the `SummaryConfigNode` class like the other processing functions then you only need to change the information that actually differs between each node (i.e., the `.number()`).

Reply:
Well - it is not exactly expensive - in the great scheme of things. The point was that the current approach allowed a more local/compact approach - which I liked. I'll take a look though.